tests/SpecTests/ClientSideEncryption/Prose25_LookupTest.php

```diff
@@ -0,0 +1,379 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests\ClientSideEncryption;
+
+use MongoDB\BSON\Binary;
+use MongoDB\Client;
+use PHPUnit\Framework\Attributes\Group;
+
+use function base64_decode;
+use function file_get_contents;
+
+/**
+ * Prose test 25: Test lookup
+ *
+ * @see https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.md#25-test-lookup
+ */
+#[Group('csfle')]
+class Prose25_LookupTest extends FunctionalTestCase
+{
+    private Client $encryptedClient;
+    private Client $unencryptedClient;
+    private $key1Id;
+
+    private const COLL_CSFLE = 'csfle';
+    private const COLL_CSFLE2 = 'csfle2';
+    private const COLL_QE = 'qe';
+    private const COLL_QE2 = 'qe2';
+    private const COLL_NO_SCHEMA = 'no_schema';
+    private const COLL_NO_SCHEMA2 = 'no_schema2';
+
+    private static string $dataDir = __DIR__ . '/../../specifications/source/client-side-encryption/etc/data/lookup';
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        if ($this->isStandalone()) {
+            $this->markTestSkipped('Lookup tests require replica sets');
+        }
+
+        $this->skipIfServerVersion('<', '7.0.0', 'Lookup encryption tests require MongoDB 7.0 or later');
+
+        $key1Document = $this->decodeJson(file_get_contents(self::$dataDir . '/key-doc.json'));
+        $this->key1Id = $key1Document->_id;
+
+        $this->unencryptedClient = self::createTestClient();
+
+        $autoEncryptionOpts = [
+            'keyVaultNamespace' => 'keyvault.datakeys',
+            'kmsProviders' => ['local' => ['key' => new Binary(base64_decode(self::LOCAL_MASTERKEY))]],
+        ];
+
+        $this->encryptedClient = self::createTestClient(null, [], [
+            'autoEncryption' => $autoEncryptionOpts,
+            /* libmongocrypt caches results from listCollections. Use a new
+             * client in each test to ensure its encryptedFields is applied. */
+            'disableClientPersistence' => true,
+        ]);
+
+        // Drop the key vault collection and insert key1Document with a majority write concern
+        self::insertKeyVaultData($this->encryptedClient, [$key1Document]);
+
+        $this->refreshCollections();
+    }
+
+    private function refreshCollections(): void
+    {
+        $db = $this->encryptedClient->selectDatabase(self::getDatabaseName());
```

Comment:
`selectDatabase` and `selectCollection` are deprecated.
You can also initialize the unencrypted Database.
```suggestion
        $encryptedDatabase = $this->encryptedClient->getDatabase(self::getDatabaseName());
        $unencryptedDatabase = $this->unencryptedClient->getDatabase(self::getDatabaseName());
```